src/ui.js

```diff
@@ -2391,6 +2425,37 @@ class UI {
         }
     }
 
+    /// Computes the size of the diagram.
+    diagram_size() {
+        let [width, height] = [0, 0];
+        // Compute the extrema of the diagram.
+        const [[x_min, y_min], [x_max, y_max]] = this.quiver.bounding_rect();
+        // Sum to compute width and height.
+        for (let x = x_min; x <= x_max; ++x) {
+            width += this.cell_size(this.cell_width, x);
```

Comment:
@varkor Actually, reading this again, I don't think I've done this is correctly. What's the correct way to get the width of a particular column vs height of a particular column?

Reply:
This looks correct to me: `this.cell_size(this.cell_width, x)` returns the width of the column for a cell whose `cell.position.x` (the left border of the cell) is `x`. There's no way to get the height of a column, as the column has infinite height.

Reply:
Sorry, I meant height of a particular row. Basically, one of the width and height summations has to be wrong because they both call `cell_size` with varying `x` and `y`

Reply:
The widths and heights of columns and rows are stored in the maps `this.cell_width` and `this.cell_height` respectively. `this.cell_size` is just an accessor for either of these, which returns the default cell size if an explicit size has not been stored for a certain index (corresponding to a column index for `this.cell_width` and a row index for `this.cell_height`). Here, the width summation is calling `this.cell_size` on `this.cell_width` and the height summation is calling `this.cell_size` on `this.cell_height`, which will pick the correct dimension in each case.

Reply:
Ah, I see!